Assets/Scripts/Turret/Shooting.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class Shooting : MonoBehaviour
+    {
+        ParticleSystem ps;
+        Ray ray;
+        RaycastHit hit;
+        Tracking tracking;
+        TurretStats turretStats;
+        //public float fireRate = 2f;
+        //public float power = 12.5f;
+        bool canFire = true;
+        void Start()
+        {
+            tracking = GetComponent<Tracking>();
+            ps = GetComponent<ParticleSystem>();
+            turretStats = GetComponent<TurretStats>();
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            ray = new Ray(transform.position, transform.forward);
+            if(Physics.Raycast(ray, out hit))
```

Comment:
`RaycastHit` and `Ray` are structs, its better to declare them inside method instead of as a class property

```suggestion
            Ray ray = new Ray(transform.position, transform.forward);
            RaycastHit hit;
            if(Physics.Raycast(ray, out hit))
```

- [Ray](https://docs.unity3d.com/ScriptReference/Ray.html)
- [RaycastHit](https://docs.unity3d.com/ScriptReference/RaycastHit.html)